monai/utils/type_conversion.py

```diff
@@ -175,6 +175,13 @@ def convert_to_numpy(data, dtype: DtypeLike = None, wrap_sequence: bool = False)
     elif has_cp and isinstance(data, cp_ndarray):
         data = cp.asnumpy(data).astype(dtype, copy=False)
     elif isinstance(data, (np.ndarray, float, int, bool)):
+        # Convert into a contiguous array first if the current dtype's size is smaller than the target dtype's size.
+        # This help improve the performance because (convert to contiguous array) -> (convert dtype) is faster
+        # than (convert dtype) -> (convert to contiguous array) when src dtype (e.g., uint8) is smaller than
+        # target dtype(e.g., float32) and we are going to convert it to contiguous array anyway later in this
+        # method.
+        if data.ndim > 0 and data.dtype.itemsize < np.dtype(dtype).itemsize:
```

Comment:
CI is failing due to this call to ndim when `data` may not be a ndarray
```
monai/utils/type_conversion.py:183:12: error: Item "float" of "Union[ndarray[Any, Any], float]" has no attribute "ndim"  [union-attr]
monai/utils/type_conversion.py:183:30: error: Item "float" of "Union[ndarray[Any, Any], float]" has no attribute "dtype"  [union-attr]
```

Reply:
Thanks @grlee77 ! 
You're right, data can be `float, int, bool` so need your suggested change.